### PR TITLE
fix: warn when using `--no-bundle` with `--minify` or `--node-compat`

### PR DESCRIPTION
**Pull request by cameron-robey**
Fixes https://github.com/cloudflare/wrangler2/issues/1491

### CODE REVIEW

File: packages/wrangler/src/publish.ts [hunk 1]
```diff
@@ -276,6 +276,19 @@ See https://developers.cloudflare.com/workers/platform/compatibility-dates for m
 		);
 	}
 
+	// Warn if user tries minify or node-compat with no-bundle
+	if (props.noBundle && minify) {
+		logger.warn(
+			"`--minify` and `--no-bundle` are not supported together. If you want to minify your Worker and disable Wrangler's bundling, please minify as part of your own bundling process."
```
Comment (threepointone) [inline packages/wrangler/src/publish.ts:282]:
```suggestion
			"`--minify` and `--no-bundle` can't be used together. If you want to minify your Worker and disable Wrangler's bundling, please minify as part of your own bundling process."
```

File: packages/wrangler/src/publish.ts [hunk 2]
```diff
@@ -276,6 +276,19 @@ See https://developers.cloudflare.com/workers/platform/compatibility-dates for m
...
+		);
+	}
+
+	if (props.noBundle && nodeCompat) {
+		logger.warn(
+			"`--node-compat` and `--no-bundle` are not supported together. If you want to polyfill Node.js built-ins and disable Wrangler's bundling, please polyfill as part of your own bundling process."
```
Comment (threepointone) [inline packages/wrangler/src/publish.ts:288]:
```suggestion
			"`--node-compat` and `--no-bundle` can't be used together. If you want to polyfill Node.js built-ins and disable Wrangler's bundling, please polyfill as part of your own bundling process."
```

**Review by threepointone — COMMENTED** [review]
couple of nits, will stamp once incorporated!

**Review by threepointone — APPROVED** [review]

**Review by JacobMGEvans — APPROVED** [review]
Another winner!! Great work 🥳

**Pull request merged**